tests/unit/onyxTest.js

```diff
@@ -53,6 +53,22 @@ describe('Onyx', () => {
             expect(keys.includes(ONYX_KEYS.OTHER_TEST)).toBe(false);
         }));
 
+    it('should not provide null values to subscriber callback when set is called with null value', () => {
+        const subscriberCallback = jest.fn();
+        Onyx.connect({
+            key: ONYX_KEYS.COLLECTION.TEST_CONNECT_COLLECTION,
+            waitForCollectionCallback: true,
+            callback: subscriberCallback,
+        });
+        const itemID = 'abcd';
+        const testKey = `${ONYX_KEYS.COLLECTION.TEST_CONNECT_COLLECTION}${itemID}`;
+        return Onyx.set(testKey, 42)
```

Comment:
Should we add tests for the `merge()` and `mergeCollection()` cases?

Reply:
Done! This exercise uncovered that `merge` and `mergeCollection` weren't really behaving how I would expect